phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java

```diff
@@ -298,4 +493,132 @@ public void run() {
             }
         }
     }
+
+    /**
+     * Task runs periodically to rebuild indexes for System.Task entries.
+     *
+     */
+    public static class IndexRebuildTask extends SelfHealingTask {
+
+        public IndexRebuildTask(RegionCoprocessorEnvironment env, long timeMaxInterval) {
+            super.init(env, timeMaxInterval);
+        }
+
+        public static String[] getArgValues(boolean directApi, boolean useSnapshot, String schemaName,
+                String dataTable, String indxTable, String tenantId) {
+            final List<String> args = Lists.newArrayList();
```

Comment:
This is leaking IndexTool internals into this task. Would be better to provide an abstraction so you can call the IndexTool directly.